File: 0209-minimum-size-subarray-sum/solution.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int n = nums.length;
        int i=0;
        int j=0;
        int sum = 0;
        int minSize = Integer.MAX_VALUE;

        while(j < n) {
            sum += nums[j];

            while(sum >= target) {
                minSize = Math.min(minSize, j-i+1);
                sum -= nums[i];
                i++;
            }
            j++;
        }
        return minSize == Integer.MAX_VALUE ? 0 : minSize;
    }
}

File: 0209-minimum-size-subarray-sum/solution.swift
class Solution {
    func minSubArrayLen(_ target: Int, _ nums: [Int]) -> Int {
        var n = nums.count
        var i=0
        var j=0
        var minLength = Int.max
        var currSum = 0

        while j < n {
            currSum += nums[j]//2

            // invalid
            while currSum >= target {
                minLength = min(minLength, j-i+1)
                currSum -= nums[i]
                i += 1
            }
            j += 1
        }
        return minLength == Int.max ? 0 : minLength
    }
}
